awscrt/s3.py

```diff
@@ -657,7 +702,9 @@ def _on_finish(self, error_code, status_code, error_headers, error_body, error_o
                 error_headers=error_headers,
                 error_body=error_body,
                 error_operation_name=error_operation_name,
-                status_code=status_code)
+                status_code=status_code,
+                did_validate_checksum=did_validate_checksum,
+                checksum_validation_algorithm=checksum_validation_algorithm)
```

Comment:
you're passing `checksum_validation_algorithm` as an int right now, you need to change it into `Optional[S3ChecksumAlgorithm]`